servers/src/main/java/tachyon/master/next/filesystem/FileSystemMaster.java

```diff
@@ -344,12 +364,27 @@ public FileBlockInfo getFileBlockInfo(long fileId, int fileBlockIndex)
     }
   }
 
-  public boolean mkdirs(TachyonURI path, boolean recursive)
-      throws InvalidPathException, FileAlreadyExistException, BlockInfoException {
+  /**
+   * Create a directory at path.
+   *
+   * @param path the path of the directory
+   * @param recursive if it is true, create necessary but nonexistent parent directories, otherwise,
+   *        the parent directories must already exist
+   * @throws InvalidPathException when the path is invalid, please see documentation on
+   *         {@link InodeTree#createPath} for more details
+   * @throws FileAlreadyExistException when there is already a file at path
+   */
+  public void mkdirs(TachyonURI path, boolean recursive) throws InvalidPathException,
+      FileAlreadyExistException {
     // TODO: metrics
     synchronized (mInodeTree) {
-      mInodeTree.createPath(path, 0, recursive, true);
-      return true;
+      try {
+        mInodeTree.createPath(path, 0, recursive, true);
+      } catch (BlockInfoException bie) {
+        // Since we are creating a directory, the block size is ignored, no such exception should
+        // happen.
+        Throwables.propagate(bie);
```

Comment:
It's equal to `new RuntimeException(bie)`, the codebase seems to prefer `Throwables.propagate(bie)`.